src/main/java/wenjieBot/Storage.java

```diff
@@ -0,0 +1,165 @@
+package wenjieBot;
+
+import wenjieBot.tasks.Deadline;
+import wenjieBot.tasks.Event;
+import wenjieBot.tasks.Task;
+import wenjieBot.tasks.ToDo;
+import wenjieBot.exceptions.NoFileException;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.io.BufferedWriter;
+
+/**
+ * The Storage class handles the reading and writing of tasks to and from a file.
+ * It allows tasks to be persisted between sessions of the bot.
+ */
+public class Storage {
+
+    private String filePath;
+    private ArrayList<Task> tasks = new ArrayList<>(0);
+
+    /**
+     * Constructs a Storage object with the specified file path.
+     * The constructor attempts to read existing tasks from the file.
+     *
+     * @param filePath the path of the file where tasks are stored.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        try {
+            readTasks();
+        } catch (NoFileException e) {
+            System.out.println("Bro I can't find a file to retrieve the data, \n" +
+                    " can help lobang me and create a file pls");
+        }
+    }
+
+    /**
+     * Loads the list of tasks from storage.
+     *
+     * @return an ArrayList of Task objects that were read from the file.
+     */
+    public ArrayList<Task> load() {
+        return tasks;
+    }
+
+    /**
+     * Reads tasks from the file specified by filePath and populates the tasks list.
+     * If the file does not exist, a NoFileException is thrown.
+     *
+     * @throws NoFileException if the file does not exist.
+     */
+    public void readTasks() throws NoFileException {
+        File file = new File(filePath);
+
+        if (!file.exists()) {
+            throw new NoFileException();
+        } else {
+            try {
+                String content = new String(Files.readAllBytes(Paths.get(filePath)));
+                String[] parts = content.split("\n");
+
+                if (!content.isBlank()) {
+                    for (String stringifiedTask : parts) {
+                        boolean isDone = stringifiedTask.charAt(4) == '1';
+                        Task taskToAdd = getTask(stringifiedTask, isDone);
+                        tasks.add(taskToAdd);
+                    }
+                }
+
+            } catch (IOException e) {
+                System.out.println("An error occurred while reading the file.");
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Converts a stringified task into a Task object based on its type (ToDo, Deadline, or Event).
+     *
+     * @param stringifiedTask the string representation of the task.
+     * @param isDone whether the task is marked as done.
+     * @return the corresponding Task object.
+     */
+    private static Task getTask(String stringifiedTask, boolean isDone) {
```

Comment:
Maybe this method could be called convertStringToTask instead of getTask to make what it does clearer? Argument names look good though!